tests/TestCase/ORM/Behavior/CounterCacheBehaviorTest.php

```diff
@@ -175,6 +175,27 @@ public function testAddScope()
         $this->assertEquals(2, $after->get('posts_published'));
     }
 
+    /**
+     * @return void
+     */
+    public function testSaveWithNullForeignKey()
+    {
+        $this->comment->belongsTo('Users');
+
+        $this->comment->addBehavior('CounterCache', [
+            'Users' => [
+                'comment_count',
+            ],
+        ]);
+
+        $entity = new Entity([
+            'title' => 'Orphan comment',
+            'user_id' => null,
+        ]);
+        $this->comment->save($entity);
```

Comment:
Should this be saveOrFail? Or the assert maybe checks if saved.

Reply:
Without this fix an exception was thrown so no exception is a test enough.

Reply:
Sure,  but the fake assert could still be a useful one maybe.

Reply:
Imagine an (accidental) regression where no exception is being thrown still, but now also not saved anymore. Thus my concern.